OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/base/BaseSecurityTokenNfcActivity.java

```diff
@@ -121,48 +110,65 @@ protected void onNfcPostExecute() {
+            intent.putExtra(ViewKeyActivity.EXTRA_SECURITY_TOKEN_AID, mSecurityTokenAid);
+            intent.putExtra(ViewKeyActivity.EXTRA_SECURITY_TOKEN_USER_ID, mSecurityTokenUserId);
+            intent.putExtra(ViewKeyActivity.EXTRA_SECURITY_TOKEN_FINGERPRINTS, mSecurityTokenFingerprints);
             startActivity(intent);
         } catch (PgpKeyNotFoundException e) {
             Intent intent = new Intent(this, CreateKeyActivity.class);
-            intent.putExtra(CreateKeyActivity.EXTRA_NFC_AID, mNfcAid);
-            intent.putExtra(CreateKeyActivity.EXTRA_NFC_USER_ID, mNfcUserId);
-            intent.putExtra(CreateKeyActivity.EXTRA_NFC_FINGERPRINTS, mNfcFingerprints);
+            intent.putExtra(CreateKeyActivity.EXTRA_SECURITY_TOKEN_AID, mSecurityTokenAid);
+            intent.putExtra(CreateKeyActivity.EXTRA_SECURITY_TOKEN_USER_ID, mSecurityTokenUserId);
+            intent.putExtra(CreateKeyActivity.EXTRA_SECURITY_FINGERPRINTS, mSecurityTokenFingerprints);
             startActivity(intent);
         }
     }
 
     /**
      * Override to use something different than Notify (UI thread)
      */
-    protected void onNfcError(String error) {
+    protected void onSecurityTokenError(String error) {
         Notify.create(this, error, Style.WARN).show();
     }
 
     /**
      * Override to do something when PIN is wrong, e.g., clear passphrases (UI thread)
      */
-    protected void onNfcPinError(String error) {
-        onNfcError(error);
+    protected void onSecurityTokenPinError(String error) {
+        onSecurityTokenError(error);
     }
 
     public void tagDiscovered(final Tag tag) {
         // Actual NFC operations are executed in doInBackground to not block the UI thread
-        if(!mTagHandlingEnabled)
+        if (!mTagHandlingEnabled)
+            return;
+
+        securityTokenDiscovered(new NfcTransport(tag));
+    }
+
+    public void usbDeviceDiscovered(final UsbDevice usbDevice) {
+        // Actual USB operations are executed in doInBackground to not block the UI thread
+        if (!mTagHandlingEnabled)
+            return;
+
+        UsbManager usbManager = (UsbManager) getSystemService(Context.USB_SERVICE);
+        securityTokenDiscovered(new UsbTransport(usbDevice, usbManager));
+    }
+
+    public void securityTokenDiscovered(final Transport transport) {
+        // Actual Smartcard operations are executed in doInBackground to not block the UI thread
+        if (!mTagHandlingEnabled)
             return;
         new AsyncTask<Void, Void, IOException>() {
             @Override
             protected void onPreExecute() {
                 super.onPreExecute();
-                onNfcPreExecute();
+                onSecurityTokenPreExecute();
             }
 
             @Override
             protected IOException doInBackground(Void... params) {
                 try {
-                    handleTagDiscovered(tag);
+                    handleSmartcard(transport);
```

Comment:
yeah, I have renamed everything two times now. I decided to stick with SecurityToken as it was used already. Will rename whats left.